src/common/components/Select/Select.tsx

```diff
@@ -2,34 +2,61 @@ import React from 'react'
 import Downshift from 'downshift'
 import caretDownIcon from '../../../common/assets/icons/ico-caret-down.svg'
 import cn from 'classnames'
+import { parseFormattedNumber } from '../../utils/format'
+import { useSelectOptions } from './select.utils'
 
-export type Option = {
+export type TOption = {
   label: string
   value: string
 }
 
-export type SelectProps = {
-  options: Option[]
-  onChange?: (option: Option | null) => void
-  selected: Option | null
+export type TBaseProps = {
   placeholder?: string
   className?: string
   label?: string
+  autocomplete?: boolean
 }
 
-export default function Select({
-  options,
-  onChange,
-  selected,
-  placeholder,
-  className,
-  label,
-}: SelectProps): JSX.Element {
+export type TOptionsProps = TBaseProps & {
+  options: TOption[]
+  onChange: (option: TOption | null) => void
+  selected?: TOption | null
+}
+
+export type TRangeProps = TBaseProps & {
+  min: number
+  max: number
+  step: number
+  onChange: (value: number | null) => void
+  value: number | null
+}
+
+export type TSelectProps = TOptionsProps | TRangeProps
+
+export default function Select(props: TSelectProps): JSX.Element {
+  const { placeholder, className, label, autocomplete = false } = props
+
+  const {
+    options,
+    selected,
+    onChange,
+    onInputValueChange,
+    inputValueRef,
+  } = useSelectOptions(props)
+
+  let textClass: string
```

Comment:
Can we change the variable name? For example: `autoCompleteColor`

Reply:
yes, it should be. fixed